client/src/shared/state/Inventory.js

```diff
@@ -25,6 +25,8 @@ class Inventory {
         // Need to keep a separate list for the hotbar as things can be rearranged.
         this.hotbar = [];
 
+        this.saveItemId = {};
```

Comment:
Would prefer this to be named more explicitly about the hotbar, i.e. `savedHotbarItemIds`

Reply:
I think there is something wrong with using an object for this instead of an array.
Why is the object structure needed, with using the `itemConfig.slotIndex` for the value?
It should be enough to have an array of strings, where each value is the item ID, and the index of the value is the position in the hotbar.
So it would look like `["ghi789", "abc123", "def456", ...]` for items in the hotbar positions 1,2, 3, 4, ...

Reply:
At that point, this extra list wouldn't even be needed. Everything needed is already kept in the main `hotbar` property. When saving the hotbar to local storage, you can just loop through it to get the ids to save.
i.e. it could be a another method:
```js
getHotbarIds() {
    // Generate an array of strings from the item ids.
    return this.hotbar.map((itemConfig) => itemConfig.id);
} 
```

Reply:
If you do not use an additional data structure, when you are loading the data from localStorage you have to create a new array that is local to the scope of the function because this.hotbar = JSON.parse(loadStorageHotbar); would be invalid. Is that fine?

Reply:
It would be expected that the loading logic would not need extra state to be persisted on the top level `Inventory` object, so yes it would be fine to use it just in local scope in `loadHotbar`.

All of the inventory IDs are strings anyway. i.e. "9e7d9177-8d9b-46a4-b21f-658c95d1fc2e"

Reply:
So where you have
```js
this.saveItemId = JSON.parse(loadStorageHotbar);
this.populateInvetorytoHotbar();
```

It could be done as
```js
this.populateInvetorytoHotbar(JSON.parse(loadStorageHotbar));
```

Reply:
Alright sounds good, feel free to commit to anything else that needs changes, I will apply all the changes tonight, and do hopefully 1 more final commit and get this task complete.